src/webxdc/integration.rs

```diff
@@ -0,0 +1,88 @@
+use crate::chat::ChatId;
+use crate::context::Context;
+use crate::message::{Message, MsgId, Viewtype};
+use crate::param::Param;
+use crate::webxdc::{maps_integration, StatusUpdateItem, StatusUpdateSerial};
+use anyhow::Result;
+
+impl Message {
+    /// Mark Webxdc message shipped with the main app as a default integration.
+    pub fn set_default_webxdc_integration(&mut self) {
+        self.hidden = true;
+        self.param.set_int(Param::WebxdcIntegration, 1);
+    }
+}
+
+impl Context {
+    /// Get Webxdc instance used for optional integrations.
+    /// If there is no integration, the caller may decide to add a default one.
+    pub async fn init_webxdc_integration(
+        &self,
+        integrate_for: Option<ChatId>,
+    ) -> Result<Option<MsgId>> {
+        if let Some(instance_id) = self.sql.get_raw_config_int("webxdc_integration").await? {
```

Comment:
> Can this be added to Config? 

i tried, however, i ran into a recursion issue i did not really understand. see the comment below: "using set_config_internal() leads to recursion warning because of sync messages"

but i would not be too afraid about misspellings - it is also probable that this all will be replaced by a database if we gave more than one integration.

> It is not clear what stored there, current map integration? 

yes. 

> What happens if there is another integration other than map?

then we would probably need a table - but we're not there, and i just removed the integration_type to make clear the future is not now :)